### DISCUSSION TITLE
trade_on_close question

### DISCUSSION THREAD

**Discussion by noahberhe** [kernc/backtesting.py#956] [Q&A]
Hello!
Sorry for the Python OOP question, but hoping someone could give me a pointer?

So I've implemented a simple breakout strategy. To simplify it: say I want to enter a long if High[-1] > High[-2], would the next() function mean I'm entering my trade on the Open AFTER High[-1], ie. not on the Open of the same candle that produced the signal? 
Or would it enter the trade on Open[-1], and hence I need to change my signal to High[-2] > High[-3]?

I thought the latter might be true from looking at the code... but then looking at e.g. the crossover function in lib, it compares [-1] and [-2] to create the signal, so I thought perhaps not?

Really appreciate the advice, I've racked my head trying to sift through the code, but I'm still getting to grips with OOP.

Thanks,
Noah

**Comment by kernc** [comment]
> To simplify it: say I want to enter a long if High[-1] > High[-2], would the next() function mean I'm entering my trade on the Open AFTER High[-1], ie. not on the Open of the same candle that produced the signal?

Yes.

> Or would it enter the trade on Open[-1], and hence I need to change my signal to High[-2] > High[-3]?

No.

Think of it simply as `next()` executing in-between subsequent bars.

**Reply by noahberhe** [reply]
Thank you, and as a follow-up to that, I am sizing each trade based on a maximum 2% Portfolio Equity loss, so I need to access my current portfolio's equity (I'm using self.equity which works), and also the actual traded price, how can I access this?

And to add to that: how can I access trade_on_close, so that I can also use that as a parameter to help calculate my trade size?
